external-providers/java-external-provider/pkg/java_external_provider/filter.go

```diff
@@ -318,12 +318,14 @@ func (p *javaServiceClient) getURI(refURI string) (string, uri.URI, error) {
 		}
 		javaFileAbsolutePath = filepath.Join(filepath.Dir(sourcesFile), filepath.Dir(path), javaFileName)
 
-		cmd := exec.Command("jar", "xf", filepath.Base(sourcesFile))
-		cmd.Dir = filepath.Dir(sourcesFile)
-		err = cmd.Run()
-		if err != nil {
-			fmt.Printf("\n java error%v", err)
-			return "", "", err
+		if _, err := os.Stat(filepath.Dir(javaFileAbsolutePath)); err != nil {
+			cmd := exec.Command("jar", "xf", filepath.Base(sourcesFile))
+			cmd.Dir = filepath.Dir(sourcesFile)
+			err = cmd.Run()
+			if err != nil {
+				fmt.Printf("\n java error%v", err)
```

Comment:
wondering if we should hard-fail here?

Reply:
I would personally let the analysis continue, since it's only one JAR that failed decompressing